From f742774228ea1cef9e4e7fba20dc30fb86bc3705 Mon Sep 17 00:00:00 2001
From: Takanari Hayama <taki@igel.co.jp>
Date: Thu, 2 Feb 2012 14:30:00 +0900
Subject: [PATCH 16/73] rcarvin: fix reversed even/odd fileds.

---
 drivers/media/video/rcarvin.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/media/video/rcarvin.c b/drivers/media/video/rcarvin.c
index 0639c40..9dae55d 100644
--- a/drivers/media/video/rcarvin.c
+++ b/drivers/media/video/rcarvin.c
@@ -290,7 +290,8 @@ static void free_buffer(struct videobuf_queue *vq,
 #define VIN_VNIE_ERROR_MASK  (VIN_VNIE_CEE | VIN_VNIE_FOE)
 
 /* VnMC */
-#define VIN_VNMC_FOC         (0x00200000) /* Field Order Control */
+//#define VIN_VNMC_FOC         (0x00200000) /* Field Order Control */
+#define VIN_VNMC_FOC         (0x00000000) /* Field Order Control */
 #define VIN_VNMC_YCAL        (0x00080000) /* YCbCr-422 input data ALignment */
 #define VIN_VNMC_VUP         (0x00000400) /* Vin register UPdate control */
 
-- 
1.7.0.4

