From 7dd33258ad219b6bf9bf829c2be5ae70a11379c8 Mon Sep 17 00:00:00 2001
From: Hideki EIRAKU <hdk@igel.co.jp>
Date: Tue, 10 Apr 2012 15:23:27 +0900
Subject: [PATCH 25/73] sound: soc: rcar: workaround for a buffer overrun when suspending aplay

---
 sound/soc/rcar/sss_pcm.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sound/soc/rcar/sss_pcm.c b/sound/soc/rcar/sss_pcm.c
index d27d0c5..787f506 100644
--- a/sound/soc/rcar/sss_pcm.c
+++ b/sound/soc/rcar/sss_pcm.c
@@ -1145,6 +1145,7 @@ static snd_pcm_uframes_t sss_pointer(struct snd_pcm_substream *substream)
 	else
 		dma_paddr = readl(&dreg->dasr);
 	position = bytes_to_frames(runtime, dma_paddr - runtime->dma_addr);
+	position -= position % runtime->period_size;
 	if (position >= runtime->buffer_size)
 		position = 0;
 
-- 
1.7.0.4

