From e2871179928fcba874156ce5a015dd0dd2dbfe96 Mon Sep 17 00:00:00 2001
From: mukawa <mukawa@igel.co.jp>
Date: Mon, 11 Jun 2012 13:19:20 +0900
Subject: [PATCH 68/73] rcarvin: Fix NULL pointer access to a buffer has been already released

---
 drivers/media/video/rcarvin.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/rcarvin.c b/drivers/media/video/rcarvin.c
index bb6bb80..72dbaff 100644
--- a/drivers/media/video/rcarvin.c
+++ b/drivers/media/video/rcarvin.c
@@ -705,6 +705,10 @@ static irqreturn_t rcar_vin_irq(int irq, void *data)
 	status_of_int = vin_read(pcdev, V0INTS);
 	vin_write(pcdev, V0INTS, status_of_int);
 
+	/* nothing to do if capture status is 'STOPPED'. */
+	if (pcdev->capture_status == STOPPED)
+		goto done;
+
 	stopped = ((vin_read(pcdev, V0MS) & VIN_VNMS_CA) == 0);
 
 	if (pcdev->request_to_stop == 0) {
@@ -769,6 +773,7 @@ static irqreturn_t rcar_vin_irq(int irq, void *data)
 		complete(&pcdev->capture_stop);
 	}
 
+done:
 	spin_unlock_irqrestore(&pcdev->lock, flags);
 
 	return IRQ_HANDLED;
-- 
1.7.0.4

